src/pages/index.js

```diff
@@ -21,73 +21,229 @@ import {
   profilePicture,
   openModal,
   closePopup,
+  profileAvatarEditButton,
   options,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupWithConfirmation from "../components/PopUpWithConfirmation.js";
 
 // Data
 const initialCards = [
   {
     name: "Yosemite Valley",
     description: "Yosemite Valley",
     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/yosemite.jpg",
+    _id: "card1",
   },
   {
     name: "Lake Louise",
     description: "Lake Louise",
     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/lake-louise.jpg",
+    _id: "card12",
   },
   {
     name: "Bald Mountains",
     description: "Bald Mountains",
     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/bald-mountains.jpg",
+    _id: "card123",
   },
   {
     name: "Latemar",
     description: "Latemar",
     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/latemar.jpg",
+    _id: "card4",
   },
   {
     name: "Vanoise National Park",
     description: "Vanoise National Park",
     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/vanoise.jpg",
+    _id: "card5",
   },
   {
     name: "Lago di Braies",
     description: "Lago di Braies",
     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/lago.jpg",
+    _id: "card6",
   },
 ];
 //
+// API initialization
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f10ef152-cb9d-40bf-b92c-7c27e5b2bf36",
+    "Content-Type": "application/json",
+  },
+});
+
+// User Info
+const userInfo = new UserInfo({
+  profileTitle: document.querySelector(".profile__title"),
+  profileDescription: document.querySelector(".profile__description"),
+  profilePicture: document.querySelector(".profile__image"),
+});
+
+// Form Validators
+const profileEditFormValidator = new FormValidator(options, profileEditForm);
+profileEditFormValidator.enableValidation();
+
+const addCardFormValidator = new FormValidator(options, addCardFormElement);
+addCardFormValidator.enableValidation();
+
+const avatarEditFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```